### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by pudidic**
Added "supported types" section

### CODE REVIEW

**Comment by EricWohlstadter** [comment]
@pudidic 

This looks great. Can you add unsupported types also?  

For example, 
 UnionType: Not supported in Spark
 MapType: Planned for future support 
 IntervalType: Planned for future support

 @HyukjinKwon @dongjoon-hyun 
Do you notice any other Spark types that aren't covered in Teddy's doc?

**Comment by pudidic** [comment]
Thanks for feedback, @EricWohlstadter. I added "Unsupported Types" section, too.

File: README.md [hunk 1]
```diff
@@ -43,6 +43,37 @@ For use in Spark cluster-mode on kerberized Yarn cluster, set:
...
+
+- A Hive String, Char, Varchar column will be converted into a Spark StringType column.
+- When a Spark StringType column has maxLength metadata, it will be converted into a Hive Varchar column. Otherwise, it will be converted into a Hive String column.
+- A Hive Timestamp column will lose sub-microsecond precision when it is converted into a Spark TimestampType column. Because a Spark TimestampType column is microsecond precision, while a Hive Timestamp column is nanosecond precision.
+
+Unsupported Types
+=====
+| Spark Type           | Hive Type | Plan                        |
+| -------------------- | --------- | --------------------------- |
+| CalendarIntervalType | Interval  | Planned for future support  |
```
Comment (HyukjinKwon) [inline README.md]: I believe this is still kind of half-public in Spark but should be okay to mention I believe :-).

File: README.md [hunk 2]
```diff
@@ -43,6 +43,37 @@ For use in Spark cluster-mode on kerberized Yarn cluster, set:
...
+| MapType              | Map       | Planned for future support  |
+| N/A                  | Union     | Not supported in Spark      |
```
Comment (HyukjinKwon) [inline README.md]: I believe there's `NullType` too.

Comment (pudidic) [reply README.md]: I added NullType. Thanks for feedback!

**Comment by EricWohlstadter** [comment]
thx @pudidic

**Pull request merged**